### PR TITLE
Preprocess dartdoc

### PR DESCRIPTION
**Pull request by jonasfj**

### CODE REVIEW

File: pkg/web_app/lib/src/autocompletion.dart [hunk 1]
```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:html';
+
+final _widgets = {
+  'autocompletion': _createAutocompletionWidget,
```
Comment (isoos) [inline pkg/web_app/lib/src/autocompletion.dart:9]: should this go in as a separate PR?

Comment (jonasfj) [reply pkg/web_app/lib/src/autocompletion.dart:9]: yes, this thing is not done yet :see_no_evil:

**Review by isoos — APPROVED** [review]

File: pkg/pub_dartdoc_data/lib/dartdoc_page.dart [hunk 2]
```diff
@@ -0,0 +1,196 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
```
[the remark this answers is not in the document]
Comment (jonasfj) [reply pkg/pub_dartdoc_data/lib/dartdoc_page.dart:1]: Yeah, it could be either place... Ideally, we probably want to do this in a background process anyways, so it can move back into `app/` later.
